projects/extension/src/background/ConnectionManager.ts

```diff
@@ -1,297 +1,425 @@
-import westend from "../../public/assets/westend.json"
-import kusama from "../../public/assets/kusama.json"
-import polkadot from "../../public/assets/polkadot.json"
-import rococo from "../../public/assets/rococo.json"
-
-export const wellKnownChains: Map<string, string> = new Map<string, string>([
-  ["polkadot", JSON.stringify(polkadot)],
-  ["kusama", JSON.stringify(kusama)],
-  ["rococo", JSON.stringify(rococo)],
-  ["westend", JSON.stringify(westend)],
-])
-
-const l = logger("Extension Connection Manager")
 
-/**
- * ConnectionManager is the main class involved in managing connections from
- * content-script.  It keeps track of active chain connections and it is also
- * responsible for triggering events when the state changes for the UI to update
- * accordingly.
- */
-export class ConnectionManager extends (EventEmitter as {
-  new (): StateEmitter
-}) {
-  #client: Client | null
-  #emitScheduled = false
-  readonly #chainConnections: Map<string, ChainConnection> = new Map()
-
-  constructor(client: Client) {
-    super()
-    this.#client = client
+import {
+  ToApplication,
+  ToExtension,
+} from "@substrate/connect-extension-protocol"
+
+export interface ChainInfo<SandboxId> {
+  chainName: string
+  healthStatus?: SmoldotHealth
+  apiInfo?: {
+    chainId: string
+    sandboxId: SandboxId
   }
+}
+
+export class ConnectionManager<SandboxId> {
+  #smoldotClient: SmoldotClient = smoldotStart()
+  #sandboxes: Map<SandboxId, Sandbox> = new Map()
+  #wellKnownChains: Map<string, WellKnownChain> = new Map()
 
   /**
+   * Adds a new well-known chain to this state machine.
    *
-   *
-   * @returns all the connected chains.
+   * While it is not strictly mandatory, you are strongly encouraged to call this at the
+   * beginning and before adding any sandbox.
    */
-  get connections(): ExposedChainConnection[] {
-    return [...this.#chainConnections.values()]
-      .filter((a) => a.chainName)
-      .map((a: ChainConnection) => ({
-        chainId: a.chainId,
-        url: a.url,
-        tabId: a.tabId,
-        chainName: a.chainName,
-        healthStatus: a.healthStatus,
-      }))
-  }
+  async addWellKnownChain(
+    chainName: string,
+    spec: string,
+    databaseContent?: string,
+  ): Promise<void> {
+    const healthChecker = smHealthChecker()
+
+    const chain = await this.#smoldotClient.addChain({
+      chainSpec: spec,
+      jsonRpcCallback: (response) => {
+        healthChecker.responsePassThrough(response)
+      },
+      databaseContent,
+      potentialRelayChains: [],
+    })
+
+    const wellKnownChain: WellKnownChain = { chain, spec, healthChecker }
 
-  #emitStateChanged() {
-    // let's make sure that we don't unnecessary spam the UI with notifications
-    if (this.#emitScheduled) return
+    healthChecker.setSendJsonRpc((rq) => chain.sendJsonRpc(rq))
+    healthChecker.start(
+      (health) => (wellKnownChain.latestHealthStatus = health),
```

Comment:
I've noticed that and it's already the case before this PR.

Reply:
Is it, though? Because I'm looking at the cod of the `ConnectionManager` I can see that [it does emit on health updates](https://github.com/paritytech/substrate-connect/blob/3f585477eba1594b12ff8a33253dfb7f0ac473d3/projects/extension/src/background/ConnectionManager.ts#L261), there is also [a test](https://github.com/paritytech/substrate-connect/blob/3f585477eba1594b12ff8a33253dfb7f0ac473d3/projects/extension/src/background/ConnectionManager.test.ts#L240-L282) to ensure that it does.

Perhaps we have a bug in the UI, if it isn't updating accordingly... I haven't checked yet, but it's possible, since we don't have any tests and there is room for improvement on the UI code, but at least the `ConnectionManager` is doing its part.

Reply:
Missed that line